File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.java
class Solution {
    public char convert(String num){
        char result = ' ';
        switch(num){
            case "zero":
                result = '0';
                break;
            case "one":
                result = '1';
                break;
            case "two":
                result = '2';
                break;
            case "three":
                result = '3';
                break;
            case "four":
                result = '4';
                break;
            case "five":
                result = '5';
                break;
            case "six":
                result = '6';
                break;
            case "seven":
                result = '7';
                break;
            case "eight":
                result = '8';
                break;
            case "nine":
                result = '9';
                break;
        }
        return result;
    }
    
    public int solution(String s) {
        int answer = 0;
        String str = "";
        for(int i = 0 ; i < s.length() ; i ++ ){
            if('0' <= s.charAt(i) && s.charAt(i) <= '9'){
                str = str + s.charAt(i);
            }
            else{
                String temp ="";
                while('a' <= s.charAt(i) && s.charAt(i) <= 'z'){
                    temp = temp + s.charAt(i++);
                    if(temp.length() >= 3 && convert(temp) != ' '){
                        break;
                    }
                }
                i = i - 1;
                str = str + convert(temp);
            }
        }
        return Integer.parseInt(str);
    }
}

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.cpp
#include <string>
#include <vector>
#include <map>
#include <iostream>

using namespace std;

int solution(string s) {
    int answer = 0;
    
    map <string, char> convert;
    convert["zero"] = '0'; convert["one"] = '1'; convert["two"] = '2';
    convert["three"] = '3'; convert["four"] = '4'; convert["five"] = '5';
    convert["six"] = '6'; convert["seven"] = '7'; convert["eight"] = '8';
    convert["nine"] = '9';
    

    
    string temp = "";
    for(int i = 0 ; i < s.size() ; i ++ ){
        if('0'<= s[i] && s[i] <= '9'){
            temp += s[i];
        }
        else{
            string str = "";
            for(int j = i ; j < s.size() ; j ++ ){
                if('A' <= s[j] && s[j] <= 'Z'){
                    str += (s[j] + 32);
                }
                else{
                    str += s[j];
                }
                if(convert[str] != '\0'){
                    temp += convert[str];
                    break;
                }
            }
            i = i + (str.size() - 1);
        }
    }
    
    cout << temp;
    answer = stoi(temp);
    
    
    return answer;
}